crates/task-impls/src/transactions.rs

```diff
@@ -203,7 +201,9 @@ where
                 self.cur_view = view;
 
                 // If we are not the next leader (DA leader for this view) immediately exit
-                if !self.committee_exchange.is_leader(self.cur_view + 1) {
+                //
+                // TODO GG is it okay to use quorum_exchange.is_leader instead of committee_exchange.is_leader? If we keep committee_exchange in this task then this is the only place where it would be used.
```

Comment:
Now that we've decided to make a separate task + exchange for VID, perhaps this PR should not switch from committee exchange to quorum exchange in the transactions task.

Per the above code comment, in the future the committee exchange would be used only at this one line in the transactions task.

In the foreseeable future, committee leader == quorum leader so it doesn't really matter, so perhaps it's easier just to use the quorum exchange now that I've done the work in this PR. OTOH if committee exchange is preferred then I should revert to that before we merge this PR. @shenkeyao any opinion?

Reply:
My vote is to merge this PR as-is.

It's unlikely that we'll ever have committee leader != quorum leader. But if for some reason committee leader != quorum leader sometime in the future then one could argue that we should prefer quorum leader over committee leader here. Why? Because this code will also affect VID, and I suspect that in a world where quorum leader != committee leader we prefer that VID leader == quorum leader. Thus, in all cases we prefer to use quorum leader here anyway, so no need to bother reverting this change.